internal/service/networkfirewall/firewall_policy.go

```diff
@@ -80,6 +80,10 @@ func ResourceFirewallPolicy() *schema.Resource {
 										Required:     true,
 										ValidateFunc: verify.ValidARN,
 									},
+									"override_action": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
```

Comment:
From the [API for CreateFirewallPolicy](https://docs.aws.amazon.com/network-firewall/latest/APIReference/API_CreateFirewallPolicy.html), I noticed `action` is a nested argument within `override`. We usually try to follow the API as close as possible as there could be future API enhancements that add new nested arguments to the parent argument. Keeping to the API documentation, this argument can potentially be the following. We can also add validation for the `action` argument following the documentation [here](https://docs.aws.amazon.com/network-firewall/latest/APIReference/API_StatefulRuleGroupOverride.html).

```suggestion
									"override": {
										Type:     schema.TypeList,
										MaxItems: 1,
										Optional: true,
										Elem: &schema.Resource{
											Schema: map[string]*schema.Schema{
												"action": {
													Type:         schema.TypeString,
													Required:     true,
													ValidateFunc: validation.StringInSlice(networkfirewall.OverrideAction_Values(), false),
												},
											},
										},
									},
```